ts/src/bitmart.ts

```diff
@@ -995,50 +995,87 @@ export default class bitmart extends Exchange {
-        //     }
+        //    {
+        //        message: 'OK',
+        //        code: '1000',
+        //        trace: '5534867fdee4403aba27b34c6922da0f.123.17225867075010273',
+        //        data: {
+        //            currencies: [
+        //                {
+        //                    currency: 'GGT',
+        //                    name: 'Grape Governance Token',
+        //                    contract_address: '0x39B0E13A29c2A27ce88ceBD21262A232b0633613',
+        //                    network: 'ERC20',
+        //                    withdraw_enabled: true,
+        //                    deposit_enabled: true,
+        //                    withdraw_minsize: '42',
+        //                    withdraw_minfee: '4.18'
+        //                },
+        //                ...
+        //            ]
+        //        }
+        //    }
         //
-        const data = this.safeValue (response, 'data', {});
-        const currencies = this.safeValue (data, 'currencies', []);
+        const data = this.safeDict (response, 'data', {});
+        const currencies = this.safeList (data, 'currencies', []);
         const result: Dict = {};
         for (let i = 0; i < currencies.length; i++) {
             const currency = currencies[i];
-            const id = this.safeString (currency, 'id');
+            const id = this.safeString (currency, 'currency');
             const code = this.safeCurrencyCode (id);
-            const name = this.safeString (currency, 'name');
             const withdrawEnabled = this.safeValue (currency, 'withdraw_enabled');
             const depositEnabled = this.safeValue (currency, 'deposit_enabled');
+            const networkId = this.safeString (currency, 'network');
             const active = withdrawEnabled && depositEnabled;
-            result[code] = {
+            const fee = this.safeNumber (currency, 'withdraw_minfee');
+            const minWithdraw = this.safeNumber (currency, 'withdraw_minsize');
+            result[code] = this.safeCurrencyStructure ({
+                'info': currency, // the original payload
                 'id': id,
                 'code': code,
-                'name': name,
-                'info': currency, // the original payload
+                'name': this.safeString (currency, 'name'),
                 'active': active,
                 'deposit': depositEnabled,
                 'withdraw': withdrawEnabled,
-                'fee': undefined,
+                'fee': fee,
+                'fees': undefined,
                 'precision': undefined,
                 'limits': {
                     'amount': { 'min': undefined, 'max': undefined },
-                    'withdraw': { 'min': undefined, 'max': undefined },
+                    'withdraw': {
+                        'min': minWithdraw,
+                        'max': undefined,
+                    },
                 },
-            };
+                'networks': [
```

Comment:
@samgermain are you sure we only have 1 network per currency?

Reply:
There might be more, but were only given information about 1 network with this endpoint, and I don't believe that there is another endpoint